src/python_testing/TC_EWATERHTR_2_1.py

```diff
@@ -0,0 +1,120 @@
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json --enable-key 000102030405060708090a0b0c0d0e0f --featureSet 0x03
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --hex-arg enableKey:000102030405060708090a0b0c0d0e0f --endpoint 1 --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto --int-arg PIXIT.EWATERHTR.EM:1 PIXIT.EWATERHTR.TP:2
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+from TC_EWATERHTRBase import EWATERHTRBase
+
+logger = logging.getLogger(__name__)
+
+
+class TC_EWATERHTR_2_1(MatterBaseTest, EWATERHTRBase):
+
+    def desc_TC_EWATERHTR_2_1(self) -> str:
+        """Returns a description of this test"""
+        return "[TC-EWATERHTR-2.1] Attributes with attributes with DUT as Server\n" \
+            "This test case verifies the non-global attributes of the Water Heater Management cluster server."
+
+    def pics_TC_EWATERHTR_2_1(self):
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        return ["EWATERHTR.S"]
+
+    def steps_TC_EWATERHTR_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep("1", "Commissioning, already done",
+                     is_commissioning=True),
+            TestStep("2", "TH reads HeaterTypes attribute.",
+                     "DUT as Server replies with a WaterHeaterTypeBitmap (enum8) greater than 0x00 (at least one type supported), and less than 0x20 (no undefined types supported)."),
+            TestStep("3", "TH reads HeatDemand attribute.",
+                     "DUT as Server replies with a WaterHeaterDemandBitmap (enum8)."),
+            TestStep("4", "TH reads TankVolume attribute.",
+                     "DUT as Server replies with a uint16 value."),
+            TestStep("5", "TH reads EstimatedHeatRequired attribute.",
+                     "DUT as Server replies with an energy-mWh value."),
+            TestStep("6", "TH reads TankPercentage attribute.",
+                     "DUT as Server replies with a percent value."),
+            TestStep("7", "TH reads BoostState attribute.",
+                     "DUT as Server replies with a BoostStateEnum (enum8) value."),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_1(self):
+
+        em_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.EM']
+        tp_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.TP']
+
+        self.step("1")
+        # Commission DUT - already done
+
+        self.step("2")
+        heaterTypes = await self.read_whm_attribute_expect_success(attribute="HeaterTypes")
```

Comment:
I understand this is listed as a PICS in the test plan, but this attribute is mandatory so is it really necessary to have a PICS check here?